### ISSUE TITLE
Fix batch norm default parameters

### ISSUE THREAD

**Issue by albertz** [rwth-i6/returnn#522]
#508

New potential defaults:

* `momentum=0.1`
* `update_sample_only_in_training=True`
* `delay_sample_update=True`
* `param_version=2` (with #898)
* `masked_time` mandatory

See also other frameworks for defaults. Some references:

* [PyTorch-to-RETURNN comment](https://github.com/rwth-i6/pytorch-to-returnn/blob/0d55be95da12fbe3beaba8205a760aa5f18de2db/pytorch_to_returnn/torch/nn/modules/batchnorm.py#L90)
* [PyTorch BatchNorm1d](https://pytorch.org/docs/stable/generated/torch.nn.BatchNorm1d.html)

**Comment by albertz** [comment]
One open question here is on our `masked_time` option, which is currently `True` by default. Which is the correct thing to do. This flattens the tensor, i.e. removes any padded frames, and calculates all the statistics only on the valid frames.

However, that option makes it slower and requires more memory. (I don't really have numbers on this. Some benchmarks would be nice...)

Also, we currently do not use the fused TF op for batch norm, which would make it faster. Although we could also use it on the flattened tensor. I don't know if there is any interplay between the two things.

Also note: All other frameworks I know do not use this padding, and always use a simple fused op. E.g. PyTorch (https://github.com/pytorch/pytorch/issues/70529) and Keras (https://github.com/tensorflow/tensorflow/issues/30161).

**Comment by albertz** [comment]
One note on the confusion of `momentum`: This probably comes from the different meaning of momentum.

RETURNN batch norm:
```
mean_new = (1 - momentum) * mean_old + momentum * mean_cur_batch
```

[PyTorch batch norm](https://pytorch.org/docs/stable/generated/torch.nn.BatchNorm1d.html):
```
mean_new = (1 - momentum) * mean_old + momentum * mean_cur_batch
```

[Keras batch norm](https://keras.io/api/layers/normalization_layers/batch_normalization/):
```
mean_new = momentum * mean_old + (1 - momentum) * mean_cur_batch
```

[PyTorch optimizer](https://pytorch.org/docs/stable/generated/torch.optim.SGD.html):
```
v_new = momentum * v_old + g
```

[Keras optimizer](https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/SGD):
```
v_new = momentum * v_old - learning_rate * g
```

**Comment by albertz** [comment]
I also suggest `param_version=2` (via #898). All the automatic param conversion should work.

**Comment by albertz** [comment]
I also suggest to make `masked_time` mandatory (not having a default). While `masked_time=True` is maybe the correct thing to do, this can be much slower, and is also not consistent to how all other frameworks are doing it (like `masked_time=False`). So the user should explicitly decide on this.
